### PR TITLE
feat(search): return scores

### PR DESCRIPTION
**Pull request by dranikpg**

### CODE REVIEW

File: src/server/search/doc_index.cc [hunk 1]
```diff
@@ -57,6 +57,12 @@ const absl::flat_hash_map<string_view, search::SchemaField::FieldType> kSchemaTy
 
 }  // namespace
 
+bool SearchParams::ShouldReturnField(std::string_view field) const {
+  auto cb = [field](const auto& entry) { return entry.first == field; };
+  return !return_fields ||
```
Comment (kostasrim) [inline src/server/search/doc_index.cc:62]: You don't have to make this change but I thought you might like it so I am mentioning it here.

Basically, with `find_if` you have to do an extra check, the `== end()` part. You can do this without it with: `std::any_of` which directly returns a `bool`. Furthermore, is has other variants as well, like `none_of` and `all_of` :stuck_out_tongue:

File: src/core/search/ast_expr.h [hunk 2]
```diff
@@ -102,4 +105,9 @@ inline std::ostream& operator<<(std::ostream& os, const dfly::search::AstExpr& a
   return os;
 }
 
+inline std::ostream& operator<<(std::ostream& os, const dfly::search::AstKnnNode& ast) {
+  // os << "ast{" << ast->Debug() << "}";
```
Comment (kostasrim) [inline src/core/search/ast_expr.h:109]: This is unimplemented. Also, to not pollute the global namespace (and avoid ADL on unqualified lookups) you can define it as a friend within the class `AstKnnNode` and that way a) it does not pollute lookup tables and b) you can also get rid of inline :stuck_out_tongue:

Comment (kostasrim) [reply src/core/search/ast_expr.h:109]: p.s. you don't have to make this change now, and if you planted to have this unimplemented comment here and I will accept

File: src/server/search/doc_index.cc (hunk 1 above)
Comment (dranikpg) [reply src/server/search/doc_index.cc:62]: True, forgot about them

File: src/core/search/ast_expr.h (hunk 2 above)
Comment (dranikpg) [reply src/core/search/ast_expr.h]: True, moved both

**Comment by dranikpg** [comment]
Let's try https://github.com/dragonflydb/dragonfly/pull/1915 to make CI happy. I couldn't reproduce it locally (but found out it doesn't work with 4.x)

**Comment by dranikpg** [comment]
Finally 🙂 I had a bug... Fixed comments

**Review by kostasrim — APPROVED** [review]

**Pull request merged**